Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DriverOnboarding/Status.hs

```diff
@@ -366,24 +366,30 @@ checkIfDocumentValid merchantOperatingCityId docType category status = do
         else return True
     Nothing -> return True
 
-getProcessedDriverDocuments :: DVC.DocumentType -> Id SP.Person -> Id DM.Merchant -> Id DMOC.MerchantOperatingCity -> Flow (Maybe ResponseStatus, Maybe Text, Maybe BaseUrl)
-getProcessedDriverDocuments docType driverId _merchantId _merchantOpCityId =
+getProcessedDriverDocuments :: DVC.DocumentType -> Id SP.Person -> Id DM.Merchant -> Id DMOC.MerchantOperatingCity -> Maybe Bool -> Flow (Maybe ResponseStatus, Maybe Text, Maybe BaseUrl)
+getProcessedDriverDocuments docType driverId _merchantId _merchantOpCityId makeSelfieAadhaarPanMandatory =
   case docType of
     DVC.DriverLicense -> do
       mbDL <- DLQuery.findByDriverId driverId -- add failure reason in dl and rc
       return (mapStatus <$> (mbDL <&> (.verificationStatus)), mbDL >>= (.rejectReason), Nothing)
-    DVC.AadhaarCard -> do
-      mbAadhaarCard <- QAadhaarCard.findByPrimaryKey driverId
-      return (mapStatus . (.verificationStatus) <$> mbAadhaarCard, Nothing, Nothing)
+    DVC.AadhaarCard ->
+      if fromMaybe False makeSelfieAadhaarPanMandatory
+        then do
+          mbAadhaarCard <- QAadhaarCard.findByPrimaryKey driverId
+          return (mapStatus . (.verificationStatus) <$> mbAadhaarCard, Nothing, Nothing)
+        else return (Just VALID, Nothing, Nothing)
     DVC.Permissions -> return (Just VALID, Nothing, Nothing)
     DVC.SocialSecurityNumber -> do
       mbSSN <- QDSSN.findByDriverId driverId
       return (mapStatus <$> (mbSSN <&> (.verificationStatus)), mbSSN >>= (.rejectReason), Nothing)
-    DVC.ProfilePhoto -> checkImageValidity DVC.ProfilePhoto driverId
+    DVC.ProfilePhoto -> if fromMaybe False makeSelfieAadhaarPanMandatory then checkImageValidity DVC.ProfilePhoto driverId else return (Just VALID, Nothing, Nothing)
```

Comment:
Same only if something is uploaded let's call it as VALID or INVALID

Reply:
same reason as above.